ccmlib/dse_cluster.py

```diff
@@ -18,10 +18,12 @@
 class DseCluster(Cluster):
 
     def __init__(self, path, name, partitioner=None, install_dir=None, create_directory=True, version=None, dse_username=None, dse_password=None, dse_credentials_file=None, opscenter=None, verbose=False):
-        if dse_credentials_file:
-            self.load_credentials_from_file(dse_credentials_file)
-        else:
+        self.dse_username = None
+        self.dse_password = None
+        self.load_credentials_from_file(dse_credentials_file)
+        if dse_username is not None:
             self.dse_username = dse_username
+        if dse_password is not None:
```

Comment:
There would be a possible issue if there is no credentials file and --dse-username and --dse-password were not passed in, here is the failure behavior with explicitly setting them to None beforehand:

```
ccmlib.common.ArgumentError: Invalid url http://downloads.datastax.com/enterprise/dse-4.8.2-bin.tar.gz (underlying error is: HTTP Error 401: Unauthorized)
```

And without:

```
AttributeError: 'DseCluster' object has no attribute 'dse_username'
```

I think the first scenario is more obvious, but maybe we should assert that there was a resolved username / password beforehand.  I'll add some additional logic for that.